### ISSUE TITLE
[lx] SNMP + UDP wants IP_PKTINFO

### ISSUE THREAD

**Issue by rzezeski** [joyent/illumos-joyent#60]
### Summary ###

SNMP wants `IP_PKTINFO` when running over UDP. This is decided at
compile-time, there is no way to disable at runtime besides running
over a different protocol.

### Steps to Reproduce ###

```
# yum -y install net-snmp net-snmp-utils
# service snmpd start
Starting snmpd: Error opening specified endpoint ""
Server Exiting with code 1
                                                           [FAILED]
```

Run with debug and you'll see that it fails setting `IP_PKTINFO`
against the `udp` domain so it falls back to `udp6` which then fails
for a different reason.

```
# snmpd -D -LS0-6d -LE0-7 -p /var/run/snmpd.pid
...
netsnmp_udp: couldn't set IP_PKTINFO: Protocol not available
```

### Workaround ###

Other transports, like TCP, work just fine.

**Comment by pfmooney** [comment]
Open as [OS-4459](https://smartos.org/bugview/OS-4459).

Thanks

**Comment by pfmooney** [comment]
Fixed in 75998c1f9d4f77c4ecb22410e25e30d192374cb6.
